### PR TITLE
Fix documentation, make ISO date format standard in some places

### PR DESCRIPTION
**Pull request by schoettl**
For #933

### CODE REVIEW

File: hledger-lib/Hledger/Data/Transaction.hs [hunk 1]
```diff
@@ -132,7 +132,7 @@ Render a journal transaction as text in the style of Ledger's print command.
 Ledger 2.x's standard format looks like this:
 
 @
-yyyy/mm/dd[ *][ CODE] description.........          [  ; comment...............]
+yyyy-mm-dd[ *][ CODE] description.........          [  ; comment...............]
```
Comment (simonmichael) [inline hledger-lib/Hledger/Data/Transaction.hs]: Not correct I think, both Ledger 2 and 3 print dates with slashes.

File: hledger-lib/Hledger/Read/TimeclockReader.hs [hunk 2]
```diff
@@ -14,7 +14,7 @@ From timeclock.el 2.6:
 A timeclock contains data in the form of a single entry per line.
 Each entry has the form:
 
-  CODE YYYY/MM/DD HH:MM:SS [COMMENT]
+  CODE YYYY-MM-DD HH:MM:SS [COMMENT]
```
Comment (simonmichael) [inline hledger-lib/Hledger/Read/TimeclockReader.hs]: https://www.ledger-cli.org/3.0/doc/ledger3.html#Time-Keeping says it contains slashes, but Ledger reads slash or dash or dot, like hledger. Emacs' timeclock.el requires slashes. I tweak this comment later maybe.

File: hledger-lib/Hledger/Data/Transaction.hs (hunk 1 above)
Comment (schoettl) [reply hledger-lib/Hledger/Data/Transaction.hs]: Oh, right. But this is the documentation for the showTransaction function, so we also have to fix the sentence above my change.

File: hledger-lib/Hledger/Read/TimeclockReader.hs (hunk 2 above)
Comment (schoettl) [reply hledger-lib/Hledger/Read/TimeclockReader.hs]: I reverted this change for now, to keep the compatibility to the original doc and Emacs.

File: hledger-lib/Hledger/Data/Transaction.hs (hunk 1 above)
Comment (schoettl) [reply hledger-lib/Hledger/Data/Transaction.hs]: I changed the above part of comment.

**Pull request merged**